doc/internal/consensus/Consensus-Protocol.md

```diff
@@ -0,0 +1,80 @@
+# The consensus
+
+Before talking about consensus, we need to discuss some logistics based on how the systems can co-exist.
+
+* Communication: Distributed systems need a method to communicate between each other. Remote Procedure Calls is the mechanism using which a standalone server can talk to another. The standard Go package [RPC](https://golang.org/pkg/net/rpc/) serves us the purpose. 
```

Comment:
How does RPC perform (code-wise) in an 1-to-many scenario?

Reply:
This is changed.